agent/agent.go

```diff
@@ -94,10 +94,76 @@ type (
 		data []byte
 		err  error
 	}
+
+	// Agent corresponds to a user and is used for storing raw Conn information
+	Agent interface {
+		GetSession() session.Session
+		Push(route string, v interface{}) error
+		ResponseMID(ctx context.Context, mid uint, v interface{}, isError ...bool) error
+		Close() error
+		RemoteAddr() net.Addr
+		String() string
+		GetStatus() int32
+		Kick(ctx context.Context) error
+		SetLastAt()
+		SetStatus(state int32)
+		Handle()
+		IPVersion() string
+		SendHandshakeResponse() error
+		SendRequest(ctx context.Context, serverID, route string, v interface{}) (*protos.Response, error)
+		AnswerWithError(ctx context.Context, mid uint, err error)
+	}
+
+	// AgentFactory factory for creating Agent instances
+	AgentFactory interface {
```

Comment:
What is the rationale for creating a factory here instead of the instance directly?